crates/chain-config/src/config/state/writer.rs

```diff
@@ -157,148 +171,103 @@ impl From<ZstdCompressionLevel> for ::parquet::basic::Compression {
+    T::OwnedKey: serde::Serialize,
+{
+    #[cfg(feature = "parquet")]
+    pub fn encode_postcard(&self) -> Vec<u8> {
+        postcard::to_stdvec(self).unwrap()
+    }
+
+    pub fn encode_json(&self) -> Value {
+        serde_json::to_value(self).unwrap()
     }
+}
 
-    pub fn json(filepath: impl Into<PathBuf>) -> Self {
+impl SnapshotWriter {
+    pub fn json(dir: impl Into<PathBuf>) -> Self {
         Self {
             encoder: EncoderType::Json {
-                state_file_path: filepath.into(),
-                buffer: StateConfig::default(),
+                builder: StateConfigBuilder::default(),
             },
+            dir: dir.into(),
         }
     }
 
     #[cfg(feature = "parquet")]
     pub fn parquet(
-        files: &crate::ParquetFiles,
+        dir: impl Into<::std::path::PathBuf>,
         compression_level: ZstdCompressionLevel,
     ) -> anyhow::Result<Self> {
-        use std::path::Path;
-
-        use ::parquet::basic::Compression;
-
-        fn create_encoder<T>(
-            path: &Path,
-            compression: Compression,
-        ) -> anyhow::Result<parquet::encode::PostcardEncoder<T>>
-        where
-            parquet::encode::PostcardEncode: parquet::encode::Encode<T>,
-        {
-            let file = std::fs::File::create(path)?;
-            parquet::encode::Encoder::new(file, compression)
-        }
-
-        let compression = compression_level.into();
-
-        let crate::ParquetFiles {
-            coins,
-            messages,
-            contracts,
-            contract_state,
-            contract_balance,
-            block_height,
-            da_block_height,
-        } = files;
-
+        let dir = dir.into();
         Ok(Self {
             encoder: EncoderType::Parquet {
-                coins: create_encoder(coins, compression)?,
-                messages: create_encoder(messages, compression)?,
-                contracts: create_encoder(contracts, compression)?,
-                contract_state: create_encoder(contract_state, compression)?,
-                contract_balance: create_encoder(contract_balance, compression)?,
-                block_height: create_encoder(block_height, compression)?,
-                da_block_height: create_encoder(da_block_height, compression)?,
+                table_encoders: std::collections::HashMap::default(),
+                compression: compression_level,
+                block_height: dir.join("block_height.parquet"),
+                da_block_height: dir.join("da_block_height.parquet"),
             },
+            dir,
         })
     }
 
-    pub fn write(mut self, state_config: StateConfig) -> anyhow::Result<()> {
-        self.write_coins(state_config.coins)?;
-        self.write_contracts(state_config.contracts)?;
-        self.write_messages(state_config.messages)?;
-        self.write_contract_state(state_config.contract_state)?;
-        self.write_contract_balance(state_config.contract_balance)?;
+    pub fn write_state_config(
```

Comment:
Hmm, it seems we use this function only in tests=) Would be nice to mark it `feature = "test-helpers"`